### ISSUE TITLE
feat(vertical-pod-autoscaler): add the ability to choose the containerPort for https service in admission-controller

### ISSUE THREAD

**Issue by 5cat** [cowboysysop/charts#402]
Hi,

VPA does not work correctly in EKS when using [aws eks module](https://github.com/terraform-aws-modules/terraform-aws-eks) due to the  [aws eks module default security group rules](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/60cc6d7a14a661a6b1b6b838c42cc057ce6f87aa/node_groups.tf#L93) not allowing port `8000` from clusterapi to node, admission-controller webhook service and container uses port `8000` which causes the kube-apiserver to error since it cannot do a post request to `https://vpa-webhook.kube-system.svc:443`.
```
failed calling webhook "vpa.k8s.io": failed to call webhook: Post "https://vpa-webhook.kube-system.svc:443/?timeout=30s": context deadline exceeded
```

There are three solutions:
1. Allow the modification of `containerPort` in the admission controller deployment, so it can be changed to either `443` or `10250` which are one of the [allowed ports by default in AWS EKS terraform module](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/60cc6d7a14a661a6b1b6b838c42cc057ce6f87aa/node_groups.tf#L93) in the helm chart.
2. allowing the traffic from the cluster API to the nodes for port `8000` via the following code
```terraform
module "eks" {
  ...
  node_security_group_additional_rules = {
    ...
    ingress_cluster_vpa = {
      description                   = "Cluster to node 8000"
      protocol                      = "tcp"
      from_port                     = 8000
      to_port                       = 8000
      type                          = "ingress"
      source_cluster_security_group = true
    }
  }
}
```
3. using [helm post rendering](https://helm.sh/docs/topics/advanced/#post-rendering) to adjust the manifests before applying with the following files:
- `./kustomization.yaml`
```yaml
resources:
  - all.yaml
patches:
  - patch: |-
      - op: replace
        path: /spec/template/spec/containers/0/ports/0/containerPort
        value: 10250
    target:
      kind: Deployment
      name: '.*admission-controller'
```
- `./kustomize`
```bash
#!/bin/bash

CURRENT_DIR=$( dirname -- "$( readlink -f -- "$0"; )"; )

cat <&0 > $CURRENT_DIR/all.yaml

kubectl kustomize $CURRENT_DIR && rm $CURRENT_DIR/all.yaml
```
then in the installation use `helm install ...  --post-renderer path/to/our/kustomize/script`
or in terraform
```terraform
module "helm_install" "vpa" {
  ...
  post_render {
    binary_path = "path/to/our/kustomize/script"
  }
}
```

I prefer the first solution since it is the most flexible and doesn't require infrastructure changes. I will create a pull request soon for it.
This issue is related to #47 https://github.com/kubernetes/autoscaler/issues/2789 and https://github.com/kubernetes/autoscaler/issues/1547

**Comment by sebastien-prudhomme** [comment]
Great. I'll also do it for all other container ports in all charts when your PR will be merged.

**Comment by sebastien-prudhomme** [comment]
Fixed by #488
